core/src/main/java/hudson/model/Queue.java

```diff
@@ -2672,6 +2707,20 @@ public void run() {
         }
     }
 
+    private class BuildableRunnable implements Runnable  {
```

Comment:
as this is not static it has a reference to this queue.   
Does this cause any issues when the queue is persisted to disk?